src/test/java/eu/xenit/gradle/docker/compose/DockerComposeConventionImplTest.java

```diff
@@ -228,5 +228,72 @@ public void testFromProjectWithDockerAlfrescoPlugin() {
         assertEquals(DOCKER_IMAGE_ID, composeSettings.getEnvironment().get("SUB_PROJECT1_DOCKER_IMAGE"));
     }
 
+    @Test
+    public void testFromProjectWithEnvironmentVariable(){
+        composeConvention.fromProject("testname",project);
+        TaskCollection<DockerBuildImage> buildImageTaskCollection = project.getTasks().withType(DockerBuildImage.class);
+        assertTaskDependsOn(composeSettings.getUpTask(), dockerBuildImageProvider);
+        assertTaskDependsOn(composeSettings.getBuildTask(), dockerBuildImageProvider);
+        assertTaskDependsOn(composeSettings.getPushTask(), dockerBuildImageProvider);
+
+        runSetupImageEnvironmentWithComposeUp();
+
+        assertEquals(DOCKER_IMAGE_ID, composeSettings.getEnvironment().get("TESTNAME_BUILD_DOCKER_IMAGE"));
+    }
+
+    @Test
+    public void testFromProjectWithEnvironmentVariable1(){
+        String DOCKER_IMAGE_ID_2 = "c0ff34";
+        TaskProvider<DockerBuildImage> dockerBuildImageProvider = project.getTasks()
+                .register("buildDockerImageAlternative", DockerBuildImage.class);
+        dockerBuildImageProvider.configure(dockerBuildImage -> {
+            dockerBuildImage.getImageId().set(DOCKER_IMAGE_ID_2);
+        });
+        composeConvention.fromProject("TESTNAME2",project);
+        TaskCollection<DockerBuildImage> buildImageTaskCollection = project.getTasks().withType(DockerBuildImage.class);
+        assertTaskDependsOn(composeSettings.getUpTask(), dockerBuildImageProvider);
+        assertTaskDependsOn(composeSettings.getBuildTask(), dockerBuildImageProvider);
+        assertTaskDependsOn(composeSettings.getPushTask(), dockerBuildImageProvider);
+
+        runSetupImageEnvironmentWithComposeUp();
+
+        assertEquals(DOCKER_IMAGE_ID, composeSettings.getEnvironment().get("TESTNAME2_BUILD_DOCKER_IMAGE"));
+        assertEquals(DOCKER_IMAGE_ID_2, composeSettings.getEnvironment().get("TESTNAME2_BUILD_DOCKER_IMAGE_ALTERNATIVE"));
+    }
+
+    @Test
+    public void testFromOtherProjectWithEnvironmentVariable() {
+        Project subProject1 = ProjectBuilder.builder().withParent(project).withName("sub-project1").build();
```

Comment:
Since this is already tested in `testFromProjoectWithDockerPluginAndEnvironmentVariable`, you can remove this testcase